*** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js (5.774 s)
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()` (1 ms)
    ✓ should use `Promise.all()` (1 ms)
    ✓ should resolve when all dice settle successfully (19 ms)
    ✕ should reject with an Error when a die rolls off the table (5003 ms)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    Die 1 rolled off the table.

      81 |           offTable = true;
      82 |         }
    > 83 |         reject(new Error(`Die ${die} rolled off the table.\n`));
         |                ^
      84 |       }
      85 |
      86 |       // If the dice settles (i.e. all mandated rolls are completed) we

      at rollOnce (3-UsingAPIs/helpers/pokerDiceRoller.js:83:16)
      at Timeout.rollOnce [as _onTimeout] (3-UsingAPIs/helpers/pokerDiceRoller.js:96:26)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    Die 2 rolled off the table.

      81 |           offTable = true;
      82 |         }
    > 83 |         reject(new Error(`Die ${die} rolled off the table.\n`));
         |                ^
      84 |       }
      85 |
      86 |       // If the dice settles (i.e. all mandated rolls are completed) we

      at rollOnce (3-UsingAPIs/helpers/pokerDiceRoller.js:83:16)
      at Timeout.rollOnce [as _onTimeout] (3-UsingAPIs/helpers/pokerDiceRoller.js:96:26)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    Die 3 rolled off the table.

      81 |           offTable = true;
      82 |         }
    > 83 |         reject(new Error(`Die ${die} rolled off the table.\n`));
         |                ^
      84 |       }
      85 |
      86 |       // If the dice settles (i.e. all mandated rolls are completed) we

      at rollOnce (3-UsingAPIs/helpers/pokerDiceRoller.js:83:16)
      at Timeout.rollOnce [as _onTimeout] (3-UsingAPIs/helpers/pokerDiceRoller.js:96:26)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    Die 4 rolled off the table.

      81 |           offTable = true;
      82 |         }
    > 83 |         reject(new Error(`Die ${die} rolled off the table.\n`));
         |                ^
      84 |       }
      85 |
      86 |       // If the dice settles (i.e. all mandated rolls are completed) we

      at rollOnce (3-UsingAPIs/helpers/pokerDiceRoller.js:83:16)
      at Timeout.rollOnce [as _onTimeout] (3-UsingAPIs/helpers/pokerDiceRoller.js:96:26)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    Die 5 rolled off the table.

      81 |           offTable = true;
      82 |         }
    > 83 |         reject(new Error(`Die ${die} rolled off the table.\n`));
         |                ^
      84 |       }
      85 |
      86 |       // If the dice settles (i.e. all mandated rolls are completed) we

      at rollOnce (3-UsingAPIs/helpers/pokerDiceRoller.js:83:16)
      at Timeout.rollOnce [as _onTimeout] (3-UsingAPIs/helpers/pokerDiceRoller.js:96:26)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      62 |   });
      63 |
    > 64 |   test('should reject with an Error when a die rolls off the table', async () => {
         |   ^
      65 |     expect.assertions(3);
      66 |     expect(exported).toBeDefined();
      67 |

      at test (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:64:3)
      at Object.describe (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:9:1)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        5.899 s, estimated 6 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.